packages/backend-storage/src/action_to_resources_map.ts

```diff
@@ -0,0 +1,72 @@
+import { ResourceAccessAcceptor } from '@aws-amplify/plugin-types';
+import { StorageAction, StoragePrefix } from './types.js';
+
+/**
+ * Internal collaborating class for maintaining the relationship between an acceptor token and the access map
+ */
+export class AcceptorTokenAccessMap {
```

Comment:
These are some internal classes to abstract away all of the "create if not exists" logic of setting elements in these maps